witwidget/notebook/colab/wit.py

```diff
@@ -270,17 +316,58 @@ def set_examples(self, examples):
     if self._rendering_complete:
       # Use BroadcastChannel to allow this call to be made in a separate colab
       # cell from the cell that displays WIT.
+      channel_start_name = 'updateExamplesStart{}'.format(self.id)
+      channel_end_name = 'updateExamplesEnd{}'.format(self.id)
       channel_name = 'updateExamples{}'.format(self.id)
+      i = 0
       output.eval_js("""(new BroadcastChannel('{channel_name}')).postMessage(
+          '')""".format(channel_name=channel_start_name))
+      while True:
+        piece = self.examples[i : i + self.SLICE_SIZE]
+        output.eval_js("""(new BroadcastChannel('{channel_name}')).postMessage(
           {examples})""".format(
-              examples=json.dumps(self.examples), channel_name=channel_name))
+              examples=json.dumps(piece), channel_name=channel_name))
+        i += self.SLICE_SIZE
+        if i > len(self.examples):
+          break
+      output.eval_js("""(new BroadcastChannel('{channel_name}')).postMessage(
+          '')""".format(channel_name=channel_end_name))
       self._generate_sprite()
 
   def infer(self):
     try:
       inferences = base.WitWidgetBase.infer_impl(self)
-      output.eval_js("""inferenceCallback({inferences})""".format(
-          inferences=json.dumps(inferences)))
+      indices = inferences['inferences']['indices'][:]
+      inferences['inferences']['indices'] = []
+      res2 = []
+      if 'classificationResult' in inferences['inferences']['results'][0]:
+        res = inferences['inferences']['results'][0]['classificationResult']['classifications'][:]
+        inferences['inferences']['results'][0]['classificationResult']['classifications'] = []
+      else:
+        res = inferences['inferences']['results'][0]['regressionResult']['regressions'][:]
+        inferences['inferences']['results'][0]['regressionResult']['regressions'] = []
+      if len(inferences['inferences']['results']) > 1:
+        if 'classificationResult' in inferences['inferences']['results'][1]:
+          res2 = inferences['inferences']['results'][1]['classificationResult']['classifications'][:]
+          inferences['inferences']['results'][1]['classificationResult']['classifications'] = []
+        else:
+          res2 = inferences['inferences']['results'][1]['regressionResult']['regressions'][:]
+          inferences['inferences']['results'][1]['regressionResult']['regressions'] = []
+      i = 0
+      output.eval_js("""inferenceStartCallback({inferences})""".format(
+            inferences=json.dumps(inferences)))
+      while True:
```

Comment:
I think it would be better to convert this so a countdown by computing the number of batches and run a for loop instead of a while loop. Having while True loop can create infinite loop bugs in future.

Reply:
will look into this